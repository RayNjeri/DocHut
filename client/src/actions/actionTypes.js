export const CLOSE_DOCUMENT = 'CLOSE_DOCUMENT';

export const DOCUMENTS_REQUEST = 'DOCUMENTS_REQUEST';
export const DOCUMENTS_SUCCESS = 'DOCUMENTS_SUCCESS';
export const DOCUMENTS_FAILURE = 'DOCUMENTS_FAILURE';

export const DOCUMENTS_ADD_REQUEST = 'DOCUMENTS_ADD_REQUEST';
export const DOCUMENTS_ADD_SUCCESS = 'DOCUMENTS_ADD_SUCCESS';
export const DOCUMENTS_ADD_FAILURE = 'DOCUMENTS_ADD_FAILURE';

export const DOCUMENTS_GET_REQUEST = 'DOCUMENT_GET_REQUEST';
export const DOCUMENTS_GET_SUCCESS = 'DOCUMENT_GET_SUCCESS';
export const DOCUMENTS_GET_FAILURE = 'DOCUMENT_GET_FAILURE';

export const DOCUMENTS_UPDATE_REQUEST = 'DOCUMENT_UPDATE_REQUEST';
export const DOCUMENTS_UPDATE_SUCCESS = 'DOCUMENT_UPDATE_SUCCESS';
export const DOCUMENTS_UPDATE_FAILURE = 'DOCUMENT_UPDATE_FAILURE';

export const DOCUMENTS_DELETE_REQUEST = 'DOCUMENT_DELETE_REQUEST';
export const DOCUMENTS_DELETE_SUCCESS = 'DOCUMENT_DELETE_SUCCESS';
export const DOCUMENTS_DELETE_FAILURE = 'DOCUMENT_DELETE_FAILURE';

export const SET_DOCUMENTS_SEARCH_FILTER_REQUEST = 'SET_DOCUMENTS_SEARCH_FILTER_REQUEST';
export const SET_DOCUMENTS_SEARCH_FILTER_SUCCESS = 'SET_DOCUMENTS_SEARCH_FILTER_SUCCESS';
export const SET_DOCUMENTS_SEARCH_FILTER_FAILURE = 'SET_DOCUMENTS_SEARCH_FILTER_FAILURE';

// Authorisation
export const IS_AUTHENTICATED = 'IS_AUTHENTICATED';
export const SET_CURRENT_USER = 'SET_CURRENT_USER';

export const SIGNUP_USER = 'SIGNUP_USER';
export const SIGNUP_SUCCESS = 'SIGNUP_SUCCESS';
export const SIGNUP_FAILURE = 'SIGNUP_FAILURE';

export const LOGIN_USER = 'LOGIN_USER';
export const LOGIN_SUCCESS = 'LOGIN_SUCCESS';
export const LOGIN_FAILURE = 'LOGIN_FAILURE';

export const LOGOUT_USER = 'LOGOUT_USER';
export const LOGOUT_SUCCESS = 'LOGOUT_SUCCESS';
export const LOGOUT_FAILURE = 'LOGOUT_FAILURE';

//user actions

export const USERS_REQUEST = 'USERS_REQUSET';
export const USERS_SUCCESS = 'USERS_SUCCESS';
export const USERS_FAILURE = 'USERS_FAILURE';

export const USERS_ADD_REQUEST = 'USERS_ADD_REQUEST';
export const USERS_ADD_SUCCESS = 'USERS_ADD_SUCCESS';
export const USERS_ADD_FAILURE = 'USERS_ADD_FAILURE';

export const USERS_GET_REQUEST = 'USER_GET_REQUEST';
export const USERS_GET_SUCCESS = 'USER_GET_SUCCESS';
export const USERS_GET_FAILURE = 'USER_GET_FAILURE';

export const USERS_UPDATE_REQUEST = 'USER_UPDATE_REQUEST';
export const USERS_UPDATE_SUCCESS = 'USER_UPDATE_SUCCESS';
export const USERS_UPDATE_FAILURE = 'USER_UPDATE_FAILURE';

export const USERS_DELETE_REQUEST = 'USER_DELETE_REQUEST';
export const USERS_DELETE_SUCCESS = 'USER_DELETE_SUCCESS';
export const USERS_DELETE_FAILURE = 'USER_DELETE_FAILURE';

//role actions
export const ROLES_REQUEST = 'ROLES_REQUSET';
export const ROLES_SUCCESS = 'ROLES_SUCCESS';
export const ROLES_FAILURE = 'ROLES_FAILURE';

export const ROLES_ADD_REQUEST = 'ROLES_ADD_REQUEST';
export const ROLES_ADD_SUCCESS = 'ROLES_ADD_SUCCESS';
export const ROLES_ADD_FAILURE = 'USERS_ADD_FAILURE';

export const ROLES_GET_REQUEST = 'ROLE_GET_REQUEST';
export const ROLES_GET_SUCCESS = 'ROLE_GET_SUCCESS';
export const ROLES_GET_FAILURE = 'ROLE_GET_FAILURE';


export const ROLES_DELETE_REQUEST = 'ROLE_DELETE_REQUEST';
export const ROLES_DELETE_SUCCESS = 'ROLE_DELETE_SUCCESS';
export const ROLES_DELETE_FAILURE = 'ROLE_DELETE_FAILURE';


