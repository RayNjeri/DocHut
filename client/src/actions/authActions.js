import request from 'superagent';
import * as tokenUtils from '../utils/tokenUtils';
import * as types from './actionTypes';

export const signUpUser = user => ({ type: types.SIGNUP_USER, user });

export const logoutUser = () => ({ type: types.LOGOUT_USER });

export const logoutSuccess = () => ({
  type: types.LOGOUT_SUCCESS
});

export const logoutFailure = message => ({ type: types.LOGOUT_FAILURE, message });

export const loginSuccessful = user => ({
  type: types.LOGIN_SUCCESS,
  isFetching: false,
  isAuthenticated: true,
  token: user.token,
  user
});

export const loginFailed = message => ({
  type: types.LOGIN_FAILURE,
  isFetching: false,
  isAuthenticated: false,
  message: JSON.parse(message.text)

});

export const setCurrentUser = user => ({
  type: types.SET_CURRENT_USER,
  user
});

export const LoginUser = user => ({ type: types.LOGIN_USER, user });

/* eslint no-undef: "off"*/

export const userSignupRequest = userData => (dispatch) => {
  dispatch(signUpUser(userData));
  return (
    request
      .post('/api/user')
      .send(userData)
      .then((response) => {
        window.localStorage.setItem('token', response.body.token);
        dispatch(loginSuccessful(response.body));
      })
      .catch((error) => {
        dispatch(loginFailed(error.response));
      })
  );
};

export const login = userData => (dispatch) => {
  dispatch(LoginUser(userData));
  return (
    request
      .post('/api/user/login')
      .send(userData)
      .then((response) => {
        window.localStorage.setItem('token', response.body.token);
        dispatch(loginSuccessful(response.body));
      })
      .catch((error) => {
        dispatch(loginFailed(error.response));
      })
  );
};


export const logout = () => (dispatch) => {
  window.localStorage.removeItem('token');
  dispatch(logoutUser());
};
