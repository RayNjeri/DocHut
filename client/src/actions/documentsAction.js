import request from 'superagent';
import jwtDecode from 'jwt-decode';
import * as types from './actionTypes';
import * as tokenUtils from '../utils/tokenUtils';

import { postEndpoint, getEndpoint, deleteEndpoint } from '../API/api';

export const closeDocument = () => ({
  type: types.CLOSE_DOCUMENT
});

export const documentsRequest = () => ({
  type: types.DOCUMENTS_REQUEST
});

export const documentsSuccess = documents => ({
  type: types.DOCUMENTS_SUCCESS,
  documents
});

export const documentsFailure = documents => ({
  type: types.DOCUMENTS_FAILURE,
  documents
});

export const documentsAddRequest = documents => ({
  type: types.DOCUMENTS_ADD_REQUEST,
  documents
});

export const documentsAddSuccess = documents => ({
  type: types.DOCUMENTS_ADD_SUCCESS,
  documents
});

export const documentsAddFailure = documents => ({
  type: types.DOCUMENTS_ADD_FAILURE,
  documents
});

export const documentsGetRequest = () => ({
  type: types.DOCUMENTS_GET_REQUEST
});

export const documentsGetSuccess = documents => ({
  type: types.DOCUMENTS_GET_SUCCESS,
  documents
});

export const documentsGetFailure = documents => ({
  type: types.DOCUMENTS_GET_FAILURE,
  documents
});

export const documentsUpdateRequest = documents => ({
  type: types.DOCUMENTS_UPDATE_REQUEST,
  documents
});

export const documentsUpdateSuccess = document => ({
  type: types.DOCUMENTS_UPDATE_SUCCESS,
  document
});

export const documentsUpdateFailure = documents => ({
  type: types.DOCUMENTS_UPDATE_FAILURE,
  documents
});

export const documentsDeleteRequest = () => ({
  type: types.DOCUMENTS_DELETE_REQUEST
});

export const documentsDeleteSuccess = documentId => ({
  type: types.DOCUMENTS_DELETE_SUCCESS,
  documentId
});

export const documentsDeleteFailure = documents => ({
  type: types.DOCUMENTS_DELETE_FAILURE,
  documents
});

export const documentsSearchFilter = searchFilter => ({
  type: types.SET_DOCUMENTS_SEARCH_FILTER,
  searchFilter
});

/* eslint no-undef: "off"*/

export const listDocuments = (limit, offset) => (dispatch) => {
  dispatch(documentsRequest());
  return (
    request
      .get(`/api/document?limit=${limit}&offset=${offset}`)
      .set('x-access-token', tokenUtils.getAuthToken())
      .then((response) => {
        dispatch(documentsSuccess(response.body));
      })
      .catch((error) => {
        dispatch(documentsFailure(error));
      })
  );
};

export const createDocument = documentData => (dispatch) => {
  dispatch(documentsAddRequest(documentData));
  return (
    request
      .post('/api/document')
      .set('x-access-token', tokenUtils.getAuthToken())
      .send(documentData)
      .then((response) => {
        dispatch(documentsAddSuccess(response.body));
      })
      .catch((error) => {
        dispatch(documentsAddFailure(error.response));
      })
  );
};

export const updateDocument = documentData => (dispatch) => {
  dispatch(documentsUpdateRequest(documentData));
  return (
    request
      .put(`/api/document/${documentData.id}`)
      .set('x-access-token', tokenUtils.getAuthToken())
      .send(documentData)
      .then((response) => {
        dispatch(documentsUpdateSuccess(response.body));
      })
      .catch((error) => {
        dispatch(documentsUpdateFailure(error.response));
      })
  );
};

export const deleteDocument = documentId => (dispatch) => {
  dispatch(documentsDeleteRequest());
  return (
    request
      .delete(`/api/document/${documentId}`)
      .set('x-access-token', tokenUtils.getAuthToken())
      .then((response) => {
        dispatch(documentsDeleteSuccess(documentId));
      })
      .catch((error) => {
        dispatch(documentsDeleteFailure(error.response));
      })
  );
};

export const getDocument = documentId => (dispatch) => {
  dispatch(documentsGetRequest());
  return (
    request
      .get(`/api/document/${documentId}`)
      .set('x-access-token', tokenUtils.getAuthToken())
      .then((response) => {
        dispatch(documentsGetSuccess(response.body));
      })
      .catch((error) => {
        dispatch(documentsGetFailure(error.response));
      })
  );
};

export const searchDocument = (title) => {
  title = encodeURIComponent(title);
  return (dispatch) => {
    return getEndpoint(`/api/search/document?q=${title}`)
      .set('x-access-token', tokenUtils.getAuthToken())
      .then((res) => {
        dispatch(documentsSearchFilter(res.body));
      })
      .catch((error) => {
        dispatch(documentsSearchFilter(error));
      });
  };
};
