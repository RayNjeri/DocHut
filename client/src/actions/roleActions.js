import request from 'superagent';
import jwtDecode from 'jwt-decode';
import * as types from './actionTypes';
import * as tokenUtils from '../utils/tokenUtils';

import { postEndpoint, getEndpoint, deleteEndpoint } from '../API/api';

export const rolesRequest = () => ({
  type: types.ROLES_REQUEST
});

export const rolesSuccess = roles => ({
  type: types.ROLES_SUCCESS,
  roles
});

export const rolesFailure = roles => ({
  type: types.ROLES_FAILURE,
  roles
});

export const rolesAddRequest = roles => ({
  type: types.ROLES_ADD_REQUEST,
  roles
});

export const rolesAddSuccess = roles => ({
  type: types.ROLES_ADD_SUCCESS,
  roles
});

export const rolesAddFailure = roles => ({
  type: types.ROLES_ADD_FAILURE,
  roles
});

export const rolesGetRequest = () => ({
  type: types.ROLES_GET_REQUEST
});

export const rolesGetSuccess = roles => ({
  type: types.ROLES_GET_SUCCESS,
  roles
});

export const rolesGetFailure = roles => ({
  type: types.ROLES_GET_FAILURE,
  roles
});

export const rolesUpdateRequest = roles => ({
  type: types.ROLES_UPDATE_REQUEST,
  roles
});

export const rolesUpdateSuccess = role => ({
  type: types.ROLES_UPDATE_SUCCESS,
  role
});

export const rolesUpdateFailure = roles => ({
  type: types.ROLES_UPDATE_FAILURE,
  roles
});

export const rolesDeleteRequest = () => ({
  type: types.ROLES_DELETE_REQUEST
});

export const rolesDeleteSuccess = roles => ({
  type: types.ROLES_DELETE_SUCCESS,
  roles
});


export const rolesDeleteFailure = roles => ({
  type: types.ROLES_DELETE_FAILURE,
  roles
});


/* eslint no-undef: "off"*/

export const listroles = () => (dispatch) => {
  dispatch(rolesRequest());
  return (
    request
      .get('/api/roles')
      .set('x-access-token', tokenUtils.getAuthToken())
      .then((response) => {
        dispatch(rolesSuccess(response.body));
      })
      .catch((error) => {
        dispatch(rolesFailure(error));
      })
  );
};

export const createrole = roleData => (dispatch) => {
  dispatch(rolesAddRequest(roleData));
  return (
    request
      .post('/api/roles')
      .set('x-access-token', tokenUtils.getAuthToken())
      .send(roleData)
      .then((response) => {
        dispatch(rolesAddSuccess(response.body));
      })
      .catch((error) => {
        dispatch(rolesAddFailure(error.response));
      })
  );
};

export const updaterole = roleData => (dispatch) => {
  dispatch(rolesUpdateRequest(roleData));
  return (
    request
      .put(`/api/roles/${roleData.id}`)
      .set('x-access-token', tokenUtils.getAuthToken())
      .send(roleData)
      .then((response) => {
        dispatch(rolesUpdateSuccess(response.body));
      })
      .catch((error) => {
        dispatch(rolesUpdateFailure(error.response));
      })
  );
};

export const deleterole = roleId => (dispatch) => {
  dispatch(rolesDeleteRequest());
  return (
    request
      .delete(`/api/roles/${roleId}`)
      .set('x-access-token', tokenUtils.getAuthToken())
      .then((response) => {
        dispatch(rolesDeleteSuccess(response.body));
      })
      .catch((error) => {
        dispatch(rolesDeleteFailure(error.response));
      })
  );
};

export const getrole = roleId => (dispatch) => {
  // dispatch(rolesGetRequest());
  return (
    request
      .get(`/api/roles/${roleId}`)
      .set('x-access-token', tokenUtils.getAuthToken())
    // .then((response) => {
    //   dispatch(rolesGetSuccess(response.body));
    // })
    // .catch((error) => {
    //   dispatch(rolesGetFailure(error.response));
    // })
  );
};

