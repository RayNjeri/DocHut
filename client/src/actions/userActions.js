import request from 'superagent';
import jwtDecode from 'jwt-decode';
import * as types from './actionTypes';
import * as tokenUtils from '../utils/tokenUtils';

import { postEndpoint, getEndpoint, deleteEndpoint } from '../API/api';

export const usersRequest = () => ({
  type: types.USERS_REQUEST
});

export const usersSuccess = users => ({
  type: types.USERS_SUCCESS,
  users
});

export const usersFailure = users => ({
  type: types.USERS_FAILURE,
  users
});

export const usersAddRequest = users => ({
  type: types.USERS_ADD_REQUEST,
  users
});

export const usersAddSuccess = users => ({
  type: types.USERS_ADD_SUCCESS,
  users
});

export const usersAddFailure =users => ({
  type: types.USERS_ADD_FAILURE,
  users
});

export const usersGetRequest = () => ({
  type: types.USERS_GET_REQUEST
});

export const usersGetSuccess = users => ({
  type: types.USERS_GET_SUCCESS,
  users
});

export const usersGetFailure = users => ({
  type: types.USERS_GET_FAILURE,
  users
});

export const usersUpdateRequest = users => ({
  type: types.USERS_UPDATE_REQUEST,
  users
});

export const usersUpdateSuccess = user => ({
  type: types.USERS_UPDATE_SUCCESS,
  user
});

export const usersUpdateFailure = users => ({
  type: types.USERS_UPDATE_FAILURE,
  users
});

export const usersDeleteRequest = () => ({
  type: types.USERS_DELETE_REQUEST
});

export const usersDeleteSuccess = users => ({
  type: types.USERS_DELETE_SUCCESS,
  users
});

export const userDeleteFailure = users => ({
  type: types.USERS_DELETE_FAILURE,
  users
});

export const usersSearchFilter = searchFilter => ({
  type: types.SET_USERS_SEARCH_FILTER,
  searchFilter
});


