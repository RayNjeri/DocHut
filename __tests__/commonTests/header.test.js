import React from 'react';
import { mount, shallow } from 'enzyme';
import shallowToJSON from 'enzyme-to-json';
import configureStore from 'redux-mock-store';
import * as actions from '../../client/src/actions/authActions';
import * as types from '../../client/src/actions/actionTypes';
import ConnectedHeader, {Header} from '../../client/src/Components/common/Header';

describe('Header Component unauthenticated', () => {

  const initialState = { authReducer: { user: {}, isAuthenticated: false }};
  const mockStore = configureStore();
  let store,container,isAunthenticated,roleId, headerWrapper;

  beforeEach(()=>{
    const props = {
      user: {},
      isAuthenticated: false
    }
    store = mockStore(initialState);
    headerWrapper = shallow(<Header {...props} /> );  
  }),

  it('Should render the header component', () => {
    const tree = shallowToJSON(headerWrapper);
    expect(tree).toMatchSnapshot();
  });
});

describe('Header Component authenticated', () => {

  const user = {
    createdAt: "2017-07-10T17:08:22.941Z",
    documents: [
      {
        access: "public",
        content: "fdvfdsvsg",
        createdAt:"2017-07-13T16:35:41.850Z",
        id: 15,
        roleId: 1,
        title: "fefsdfvfds",
        updatedAt:"2017-07-13T16:35:41.850Z",
        userId: 3
      }
    ],
    email: "ray@gmail.com",
    firstName: "Rachael",
    id: 3,
    lastName: "Njeri",
    password: "$2a$10$yimKcc5xy.4YvulsFdC/pO2IADJKLHOZgZGwggtDxliQbL0K4HY5S",
    roleId: 1,
    updatedAt: "2017-07-13T06:30:25.765Z",
    userName: "Rachael",
      
  }
  let store,container,isAunthenticated,roleId, headerWrapper;

  beforeEach(()=>{
    const props = {
      user: user,
      isAuthenticated: true
    }
    headerWrapper = shallow(<Header {...props} /> );  
  }),

  it('Should render the header component with user authenticated', () => {
    const tree = shallowToJSON(headerWrapper);
    expect(tree).toMatchSnapshot();
  });
});