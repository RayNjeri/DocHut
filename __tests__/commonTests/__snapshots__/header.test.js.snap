// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component Should render the header component 1`] = `
<nav>
  <AppBar
    className="navbar"
    iconClassNameRight="muidocs-icon-navigation-expand-more"
    showMenuIconButton={false}
    style={
      Object {
        "background": "#3bac95",
      }
    }
    title="DocHut"
    titleStyle={
      Object {
        "flex": "none",
      }
    }
    zDepth={1}
  >
    <span>
      <IndexLink
        activeClassName="active"
        to="/"
      >
        Home
      </IndexLink>
    </span>
  </AppBar>
</nav>
`;

exports[`Header Component authenticated Should render the header component 1`] = `undefined`;

exports[`Header Component authenticated Should render the header component with user authenticated 1`] = `
<nav>
  <AppBar
    className="navbar"
    iconClassNameRight="muidocs-icon-navigation-expand-more"
    showMenuIconButton={false}
    style={
      Object {
        "background": "#3bac95",
      }
    }
    title="DocHut"
    titleStyle={
      Object {
        "flex": "none",
      }
    }
    zDepth={1}
  >
    <span>
      <IndexLink
        activeClassName="active"
        to="/"
      >
        Home
      </IndexLink>
    </span>
    <span>
      <Link
        activeClassName="active"
        onlyActiveOnIndex={false}
        style={Object {}}
        to="/users"
      >
        Users
      </Link>
      <Link
        activeClassName="active"
        onlyActiveOnIndex={false}
        style={Object {}}
        to="/roles"
      >
        Roles
      </Link>
    </span>
    <span>
      <Link
        activeClassName="active"
        onlyActiveOnIndex={false}
        style={Object {}}
        to="/documents"
      >
        Documents
      </Link>
      <Link
        activeClassName="active"
        onlyActiveOnIndex={false}
        style={Object {}}
        to="/profile"
      >
        Profile
      </Link>
      <Link
        activeClassName="active"
        onClick={[Function]}
        onlyActiveOnIndex={false}
        style={Object {}}
        to="/"
      >
        Logout
      </Link>
    </span>
  </AppBar>
</nav>
`;

exports[`Header Component unauthenticated Should render the header component 1`] = `
<nav>
  <AppBar
    className="navbar"
    iconClassNameRight="muidocs-icon-navigation-expand-more"
    showMenuIconButton={false}
    style={
      Object {
        "background": "#3bac95",
      }
    }
    title="DocHut"
    titleStyle={
      Object {
        "flex": "none",
      }
    }
    zDepth={1}
  >
    <span>
      <IndexLink
        activeClassName="active"
        to="/"
      >
        Home
      </IndexLink>
    </span>
  </AppBar>
</nav>
`;
